### DISCUSSION TITLE
What's bring naming `keepUnusedDataFor` instead of `CacheLifetime`?

### DISCUSSION THREAD

**Discussion by ryota-murakami** [reduxjs/redux-toolkit#1514] [Q&A] [answered]
I felt a little bit hard learning RTKQuery first time, but now becoming gradually familiar now I'm really like it RTKQuery!
I'm afraid telling annoying question What's bring naming `keepUnusedDataFor` ?

Generally this parameter named like `CacheLifetime`, `ExpireDate`, so I feels like unique naming about `keepUnusedDataFor`.

**Comment by phryneas** [comment, marked as answer]
Hmm. Good question. No idea why I chose *this* honestly. 

`CacheLifetime` or `ExpireDate` would both not describe what it does - they key is the "unused" aspect here - keeping data around after no component has subscribed to it any more. `CacheLifetime` would imply a lifetime after something has entered the cache - but as long as it is subscribed, there will not be any expiration. Same goes for `expireDate` - that is a fixed date in the future. As long as that data is being used, we don't know it yet.

So it is the time that unused data is still being kept in the cache - as an option: `{ keepUnusedDataFor: 60 /* seconds */ }`.

**Reply by ryota-murakami** [reply]
Thank you for your great answer!
I really satisfied with this answer and `keepUnusedDataFor` is so make sense term I think!
